crates/fuel-core/src/graphql_api/da_compression.rs

```diff
@@ -314,13 +310,14 @@ where
             // This is a genesis coin, which is handled differently.
             // See CoinConfigGenerator::generate which generates the genesis coins.
             let mut bytes = [0u8; 32];
-            let tx_index = c.tx_pointer.tx_index();
-            bytes[..std::mem::size_of_val(&tx_index)]
-                .copy_from_slice(&tx_index.to_be_bytes());
-            return Ok(fuel_core_types::fuel_tx::UtxoId::new(
+            bytes[..size_of::<u16>()].copy_from_slice(&c.output_index.to_be_bytes());
+
+            let utxo_id = fuel_core_types::fuel_tx::UtxoId::new(
                 fuel_core_types::fuel_tx::TxId::from(bytes),
-                0,
-            ));
+                c.output_index,
+            );
```

Comment:
It would be nice to use the same function here and in `CoinConfigGenerator`. Could you reuse it in both places please to connect them?

Reply:
addressed in [e9eb633](https://github.com/FuelLabs/fuel-core/pull/2593/commits/e9eb633a3323fa292f310f0233f3d9d90cae77f3)